### PR TITLE
Refactored procuration matching

### PR DESCRIPTION
**Pull request by HeahDude**

### CODE REVIEW

File: src/Procuration/Filter/ProcurationProxyProposalFilters.php [hunk 1]
```diff
@@ -37,20 +37,20 @@ public function apply(QueryBuilder $qb, string $alias): void
 
         $status = $this->getStatus();
         if (self::UNASSOCIATED === $status) {
-            $qb->andWhere(sprintf('%s.foundRequest IS NULL', $alias));
+            $qb->andWhere("$alias.foundRequest IS NULL");
```
Comment (thomasbisignani) [inline src/Procuration/Filter/ProcurationProxyProposalFilters.php]: Why remove these `sprintf` ?

File: templates/procuration/_request_reasons.html.twig [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<div class="profile-value" id="request-reason">
+    {% if request.reason == constant('REASON_RESIDENCY', request) %}
```
Comment (thomasbisignani) [inline templates/procuration/_request_reasons.html.twig]: What do you think to refactor this elseif suite in one line with translations, for example :
`('procuration.request.reason.' ~ request.reason)|trans`

File: src/Procuration/Filter/ProcurationProxyProposalFilters.php (hunk 1 above)
Comment (Remg) [reply src/Procuration/Filter/ProcurationProxyProposalFilters.php]: @thomasbisignani https://github.com/EnMarche/en-marche.fr/pull/1943#discussion_r159080614

File: templates/procuration/_request_reasons.html.twig (hunk 2 above)
Comment (Remg) [reply templates/procuration/_request_reasons.html.twig]: what is the reason is unknown or not translated ? the current logic does not provide a fallback label

Comment (HeahDude) [reply templates/procuration/_request_reasons.html.twig]: This is a choice with a constraint, no need for a fallback.

Comment (HeahDude) [reply templates/procuration/_request_reasons.html.twig]: Good call, done in https://github.com/EnMarche/en-marche.fr/pull/1977/commits/b5fe1fd7370a3e893082c5b6fb6d2e0fdab5c53e.

**Pull request merged**